deepchem/models/layers.py

```diff
@@ -471,18 +471,83 @@ def call(self, inputs):
 
 
 def _cosine_dist(x, y):
-  """Computes the inner product (cosine distance) between two tensors.
+  """Computes the inner product (cosine similarity) between two tensors.
```

Comment:
This docstring is great! Given the improved docs, we might as well go ahead and make this function public 
```
def cosine_dist(x, y)
```
Could you also add the public function to `deepchem/docs/layers.rst`? This should look like
```
.. autofunction:: deepchem.models.layers.cosine_dist
```